quipucords/scanner/openshift/api.py

```diff
@@ -138,6 +140,30 @@ def retrieve_cluster(self, **kwargs) -> OCPCluster:
         cluster_entity = self._init_cluster(clusters[0])
         return cluster_entity
 
+    def retrieve_pods(self, **kwargs) -> List[OCPPod]:
+        """Retrieve OCP Pods."""
+        pods_raw = self._list_pods(**kwargs)
+        pods_list = []
+        for pod in pods_raw.items:
+            ocp_pod = OCPPod.from_api_object(pod)
+            pods_list.append(ocp_pod)
+        return pods_list
+
+    def retrieve_apps(self, **kwargs):
+        """Retrieve OCP "Apps"."""
+        pod_list = self.retrieve_pods()
```

Comment:
There is, but for oc "app" has a more strict meaning and wouldn't cover what I'm trying to collect here.
Out of ignorance I was collecting "deployments" thinking this would give us all possible containers that could be created in a ocp/k8s cluster w/o all repetition for pods with more than one replica.

I chose the word "app" here because it is how it appears on metadata labels.

I also thought on calling it  **CONTROLLER** instead of **APP** based on this piece from k8s API docs:

># **WORKLOADS**
> 
> Workloads resources are responsible for managing and running your containers on the cluster. [Containers](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#container-v1-core) are created by Controllers through [Pods](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#pod-v1-core). Pods run Containers and provide environmental dependencies such as shared or persistent storage [Volumes](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#volume-v1-core) and [Configuration](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#configmap-v1-core) or [Secret](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#secret-v1-core) data injected into the container.
> The most common Controllers are:
> -   [Deployments](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#deployment-v1-apps) for stateless persistent apps (e.g. HTTP servers).
> -   [StatefulSets](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#statefulset-v1-apps) for stateful persistent apps (e.g. databases).
> -   [Jobs](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#job-v1-batch) for run-to-completion apps (e.g. batch Jobs).

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#-strong-workloads-apis-strong-

Do you think adopting the nomenclature "controller" could be less confusing?

Reply:
There is, but for oc "app" has a more strict meaning and wouldn't cover what I'm trying to collect here.
Out of ignorance I was collecting "deployments" thinking this would give us all possible containers that could be created in a ocp/k8s cluster w/o all repetition for pods with more than one replica.

I chose the word "app" here because it is how it appears on metadata labels.

I also thought on calling it  **CONTROLLER** instead of **APP** based on this piece from k8s API docs:
>># **WORKLOADS**
> 
> Workloads resources are responsible for managing and running your containers on the cluster. [Containers](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#container-v1-core) are created by Controllers through [Pods](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#pod-v1-core). Pods run Containers and provide environmental dependencies such as shared or persistent storage [Volumes](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#volume-v1-core) and [Configuration](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#configmap-v1-core) or [Secret](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#secret-v1-core) data injected into the container.
> The most common Controllers are:
> -   [Deployments](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#deployment-v1-apps) for stateless persistent apps (e.g. HTTP servers).
> -   [StatefulSets](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#statefulset-v1-apps) for stateful persistent apps (e.g. databases).
> -   [Jobs](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#job-v1-batch) for run-to-completion apps (e.g. batch Jobs).

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#-strong-workloads-apis-strong-

Do you think adopting the nomenclature "controller" could be less confusing?

Reply:
Ahh I see.  Naming as App is confusing then. I'm not crazy about naming as controller as that's too close to Kubernetes Go operators/controllers.  Maybe name them `Workload` with an object type (i.e. Deployment, Job, ReplicaSet, etc) ? Naming is hard.